types/react-router-navigation-core/index.d.ts

```diff
@@ -0,0 +1,157 @@
+// Type definitions for react-router-navigation-core 1.0
+// Project: https://github.com/LeoLeBras/react-router-navigation#readme
+// Definitions by: Kalle Ott <https://github.com/kaoDev>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.4
+
+// High-level wrappers
+import { PureComponent, ReactNode, ComponentClass, ReactElement } from "react";
+import { BackHandler, StyleProp, ViewStyle } from "react-native";
+import { History, Location } from "history";
+import { RouterProps, RouteProps, match } from "react-router";
+
+export type Route<T = {}> = {
+    key: string;
+    routeName: string;
+    match?: match<T>;
+};
+
+export type NavigationState<OwnRoute> = {
+    index: number;
+    routes: Array<Route & OwnRoute>;
+};
+
+export type RouteProps = {
+    component?: ComponentClass<RouterProps>;
+    render?: (props: RouterProps) => ReactNode;
+    children?: ((props: RouterProps) => ReactNode) | ReactNode;
+    path?: string;
+    exact?: boolean;
+    strict?: boolean;
+};
+
+export type Card = RouteProps & {
```

Comment:
It's better to use interface inheritance instead of intersection types.

Reply:
what are the exact benefits of interfaces vs types? I personally prefer the type keyword because it is close to other functional languages.